### PR TITLE
User Status Grouping and Name Sorting.

### PR DESCRIPTION
**Pull request by DavidChristiansen**

### CODE REVIEW

File: JabbR/Chat.ui.js [hunk 1]
```diff
@@ -223,19 +228,105 @@
...
+                }
+            } else {
+                if (!this.appearsInList($user, this.idleusers)) {
+                    this.addUserToList($user, this.idleusers);
+                }
+            }
+        };
+
+        this.sortLists = function () {
+            this.sortList(this.activeusers);
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: Camel casing => activeUsers

File: JabbR/Chat.ui.js [hunk 2]
```diff
@@ -223,19 +228,105 @@
...
+            this.sortList(this.idleusers);
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: Same as above.

File: JabbR/Chat.ui.js [hunk 3]
```diff
@@ -223,19 +228,105 @@
...
+        };
+
+        this.sortList = function (listToSort) {
+            var listitems = listToSort.children('li').get();
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: Same here, camel casing.

File: JabbR/Chat.ui.js [hunk 4]
```diff
@@ -223,19 +228,105 @@
...
+            listitems.sort(function (a, b) {
+                var compA = $(a).data('name').toUpperCase();
+                var compB = $(b).data('name').toUpperCase();
+                return (compA < compB) ? -1 : (compA > compB) ? 1 : 0;
+            })
+            $.each(listitems, function (idx, itm) { listToSort.append(itm); });
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: `(index, item)` please

File: JabbR/Chat.ui.js [hunk 5]
```diff
@@ -294,10 +386,28 @@
                               .appendTo($chatArea)
                               .hide();
 
-
-        $('<ul/>').attr('id', 'users-' + roomId)
-                  .addClass('users')
-                  .appendTo($chatArea).hide();
+        if (roomName != "lobby"){
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: `!==`

File: JabbR/Chat.ui.js [hunk 6]
```diff
@@ -466,10 +575,14 @@
     function updateNote(userViewModel, $user) {
         var $note = $user.find('.note'),
             noteText = userViewModel.note,
-            noteTextUencoded = null;
+            noteTextUencoded = null,
+            requireroomUpdate = false;
```
Comment (davidfowl) [inline JabbR/Chat.ui.js]: Camel casing again :)

**Review by Scutua — APPROVED** [review]

**Pull request merged**